voxblox/include/voxblox/utils/layer_utils.h

```diff
@@ -111,7 +116,61 @@ bool isSameVoxel(const EsdfVoxel& voxel_A, const EsdfVoxel& voxel_B);
 template <>
 bool isSameVoxel(const OccupancyVoxel& voxel_A, const OccupancyVoxel& voxel_B);
 
+// This function will shift all the blocks such that the new grid origin will be
+// close to the centroid of all allocated blocks. The new_layer_origin is the
+// origin of the new grid expressed in the old grids coordinate frame.
+template <typename VoxelType>
+void centerBlocksOfLayer(Layer<VoxelType>* layer, Point* new_layer_origin) {
+  CHECK_NOTNULL(layer);
+  CHECK_NOTNULL(new_layer_origin);
+
+  // Compute the exact cenroid of all allocated block indices.
+  Point centroid = Point::Zero();
+  BlockIndexList block_indices;
+  layer->getAllAllocatedBlocks(&block_indices);
+  for (const BlockIndex block_index : block_indices) {
+    centroid += centroid;
```

Comment:
wait am I missing something or is centroid still = Point::Zero() at this stage, making this do nothing?

Reply:
hahaha I think you are right... This function currently doens't do anything. Good catch.  I guess I was just too tired :P

Reply:
I can't believe how I could fuck this up even with a unit test.... I just set the expected returned offset 
 values to 0 to see what the output would be the first time I run it. However the function returned 0 offset and the unit test passed and I probably got distracted in the meantime, so when I continued it all seemed in order -> Done :D